merlin/models/tf/models/base.py

```diff
@@ -736,6 +736,15 @@ def train_step(self, data):
         with tf.GradientTape() as tape:
             x, y, sample_weight = unpack_x_y_sample_weight(data)
 
+            # Ensure that we don't have any ragged or sparse tensors passed at training time.
+            if isinstance(x, dict):
+                for k in x:
+                    if isinstance(x[k], (tf.RaggedTensor, tf.SparseTensor)):
+                        raise ValueError(
+                            "Training with RaggedTensor or SparseTensor inputs is unsupported. "
```

Comment:
```suggestion
                            "Training with RaggedTensor or SparseTensor inputs is supported "                            
                            "for training and evaluation, but not for inference "
                            "(with Triton Inference Server or TFServing) because the "
                            "input feature names are lost in SavedModel when ragged "
                            and sparse tensors are fed as input."
```